packages/Meshfree/test/tstNearestNeighborExodusGenerator.cpp

```diff
@@ -0,0 +1,434 @@
+        "src_coords_device", num_src, space_dim );
+    auto src_coord_copy = KOKKOS_LAMBDA( const size_t i )
+    {
+        for ( size_t n = 0; n < space_dim; ++n )
+            src_coords_device( i, n ) = src_coords( i, n );
+    };
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, num_src ),
+                          src_coord_copy );
+    Kokkos::fence();
+
+    Kokkos::View<DataTransferKit::Coordinate **, Device> tgt_coords_device(
+        "tgt_coords_device", num_tgt, space_dim );
+    auto tgt_coord_copy = KOKKOS_LAMBDA( const size_t i )
+    {
+        for ( size_t n = 0; n < space_dim; ++n )
+            tgt_coords_device( i, n ) = tgt_coords( i, n );
+    };
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, num_tgt ),
+                          tgt_coord_copy );
+    Kokkos::fence();
+
+    // Create a nearest neighbor operator.
+    DataTransferKit::NearestNeighborOperator<Device> nearest_op(
+        comm, src_coords_device, tgt_coords_device );
+
+    // For now we copy the source field and create a target field with views
+    // that use the default layout for the device type. Also note that we are
+    // only using a single field dimension to be compatible with the current
+    // operator definition.
+    Kokkos::View<double *, Device> src_field_device( "src_field_device",
+                                                     num_src );
+    auto src_field_copy = KOKKOS_LAMBDA( const size_t i )
+    {
+        src_field_device( i ) = src_field( i, 0 );
+    };
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, num_src ),
+                          src_field_copy );
+    Kokkos::fence();
+
+    Kokkos::View<double *, Device> tgt_field_device( "tgt_field_device",
+                                                     num_tgt );
+
+    // Apply the operator.
+    nearest_op.apply( src_field_device, tgt_field_device );
+
+    // Create the expected nearest coordinates.
+    Kokkos::View<DataTransferKit::Coordinate **, Kokkos::LayoutLeft,
+                 Kokkos::Serial>
+        nearest_src_coords;
+    computeNeighborsBruteForce( comm, src_coords, tgt_coords,
+                                nearest_src_coords );
+
+    // Check that the field evaluated at the points found via brute-force
+    // search is the same as those computed by the operator.
+    test_result = Kokkos::View<int **, Kokkos::LayoutLeft, Kokkos::Serial>(
+        "test_result", num_tgt, 1 );
+    auto host_tgt_field = Kokkos::create_mirror_view( tgt_field_device );
+    Kokkos::deep_copy( host_tgt_field, tgt_field_device );
+    double data_val;
+    for ( size_t t = 0; t < num_tgt; ++t )
+    {
+        data_val =
+            data_func( nearest_src_coords( t, 0 ), nearest_src_coords( t, 1 ),
+                       nearest_src_coords( t, 2 ), 0 );
+        test_result( t, 0 ) =
+            std::abs( host_tgt_field( t ) - data_val ) < 1.0e-12;
+    }
```

Comment:
Why wouldn't you just do your assertions in here?

Reply:
I agree but it doesn't compile:
```
[ 98%] Building CXX object DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/tstNearestNeighborExodusGenerator.cpp.o
/scratch/source/trilinos/release/DataTransferKit/packages/Meshfree/test/tstNearestNeighborExodusGenerator.cpp(267): error: identifier "out" is undefined
          detected during instantiation of "void ExodusProblemGenerator_one_to_one_UnitTest<Node>::runUnitTestImpl(Teuchos::FancyOStream &, __nv_bool &) const [with Node=Kokkos_Compat_KokkosSerialWrapperNode]" 
(403): here

/scratch/source/trilinos/release/DataTransferKit/packages/Meshfree/test/tstNearestNeighborExodusGenerator.cpp(267): error: identifier "success" is undefined
          detected during instantiation of "void ExodusProblemGenerator_one_to_one_UnitTest<Node>::runUnitTestImpl(Teuchos::FancyOStream &, __nv_bool &) const [with Node=Kokkos_Compat_KokkosSerialWrapperNode]" 
(403): here

2 errors detected in the compilation of "/tmp/tmpxft_0000292c_00000000-7_tstNearestNeighborExodusGenerator.cpp1.ii".
DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/build.make:62: recipe for target 'DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/tstNearestNeighborExodusGenerator.cpp.o' failed
make[2]: *** [DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/tstNearestNeighborExodusGenerator.cpp.o] Error 2
CMakeFiles/Makefile2:5514: recipe for target 'DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/all' failed
make[1]: *** [DataTransferKit/packages/Meshfree/test/CMakeFiles/DataTransferKitMeshfree_NearestNeighborExodusGenerator.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

Ultimately I think this is due to the fact that this is a templated unit test and the macros just can't figure it out.

Reply:
Add the `Teuchos::FancyOStream &out` and `bool &success` arguments to your function.